keanu-project/src/main/java/io/improbable/keanu/algorithms/variational/BoundsCalculator.java

```diff
@@ -0,0 +1,75 @@
+package io.improbable.keanu.algorithms.variational;
+
+import io.improbable.keanu.tensor.TensorShape;
+import io.improbable.keanu.tensor.dbl.DoubleTensor;
+import io.improbable.keanu.vertices.Vertex;
+import lombok.RequiredArgsConstructor;
+import org.apache.commons.math3.optim.SimpleBounds;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+@RequiredArgsConstructor
+public class BoundsCalculator {
+
+    private final double boundsRange;
+    private final OptimizerBounds optimizerBounds;
+
+    public SimpleBounds getBounds(List<? extends Vertex<DoubleTensor>> latentVertices, double[] startPoint) {
+        List<Double> minBounds = new ArrayList<>();
+        List<Double> maxBounds = new ArrayList<>();
+
+        for (Vertex<DoubleTensor> vertex : latentVertices) {
+
+            if (optimizerBounds.hasBound(vertex)) {
+                validateBoundsForVertex(vertex, optimizerBounds);
+                addBoundsForVertex(vertex, optimizerBounds, minBounds, maxBounds);
```

Comment:
Should we drop the OptimizerBounds parameter from these two methods given we always just pass the member through to them?

Reply:
You good catch. It's a slip on the refactoring from pure functions.